src/Halogen/Hooks/Extra/Hooks/UseStateFn.purs

```diff
@@ -44,9 +55,83 @@ useStateFn
 useStateFn fn initial = Hooks.wrap Hooks.do
   map fn <$> useState initial
 
-useModifyState
+-- | Just like `useState`, but provides a convenience function for updating
+-- | state, rather than a state index to pass to `Hooks.modify_`.
+-- |
+-- | Example:
+-- | ```
+-- | count /\ modifyCount <- useModifyState_ 42
+-- | modifyCount (add 1)
+-- | ```
+-- |
+-- | Instead of:
+-- | ```
+-- | count /\ countIdx <- Hooks.useState 42
+-- | Hooks.modify_ countIdx (add 1)
```

Comment:
This may not be necessary to note, but this code is not valid because you can't run arbitrary `HookM` code in the body of a Hook. Valid examples of this would be:

```purs
count /\ countId <- Hooks.useState 42

let
  onClick :: HookM m Unit
  onClick = Hooks.modify_ countId (add 1)
```

or

```purs
count /\ countId <- Hooks.useState 42

Hooks.useLifecycleEffect do
  Hooks.modify_ countId (add 1)
  pure Nothing
```

but you can't do

```purs
count /\ countId <- Hooks.useState 42
Hooks.modify_ countId (add 1) -- Error here
```

Reply:
Yes, I was trying to keep these examples concise. Maybe ellipses between the two lines?

Reply:
Perhaps the ellipses in front of the code, to indicate it's not part of the same block:

```purs
count /\ countId <- Hooks.useState 42
... Hooks.modify_ countId (add 1)
```